content/blog/eng/automation-testing-a-step-back.md

```diff
@@ -0,0 +1,114 @@
+
+## Honorable mentions
+
+There is another point that usually people talk about as a benefit of working on a well-tested project: it is confidence; since developing while having your shoulders covered by tests is just mentally lighter.
+
+I didn't put it on the list because I am not comparing automation-tests with no-tests, but automation-tests with manual-tests.
+
+# Can test harm?
+
+We have seen what are the pros of automation testing, now let's examine what are the cons.
+
+Tests come at a cost: they can slow down the development process, they can slow down the CI, they can make harder maintenance and refactoring, they can make you write worst code.
+
+## Slow down the development process
+
+It has happened to me several times that writing tests has consumed me much more time and effort than developing the tested feature. It can happen due to inexperience, or due to technical limitations for which there is a greater complexity in writing tests.
+
+In these cases I always ask myself if it is really worth it, if the time invested in writing these tests will pay off, and if so after how long.
+
+So focusing only on the first of the benefits, for simplicity, we compare the time invested by writing the tests with the gain that will be obtained over time. When investing money, we can understand if the investment is good based on how long it will take to get back into the investment. The same rule applies to software automation.
+
+![enter image description here](https://imgs.xkcd.com/comics/is_it_worth_the_time_2x.png)
+
+To evaluate it, I refer to this scheme found on internet.
+
+## Slow down the CI
+
+Having a test suite makes sense when it is used in a continuous integration system. Typically tests are fairly fast when taken individually, but these can start to become a problem when they are in large quantities.
+
+This may seem like a marginal problem but having a slow CI is problematic. So be careful because at the beginning they can be like grains of sand and when the project becomes very large, the CI can become like a sandstorm and block the development process like a slow build.
+
+## Make harder maintenance and refactoring
+
+This is the case when a change to the codebase (like refactoring an implementation detail or adding a new feature) also implies changes to the test code even if the previous tested behavior doesn't change.
+
+This means that your test code is heavily coupled with your tested code. And can make frustrating, slow and inefficient working on a project.
+
+## Make write worst code
+
+This point is the exact opposite of the one written above on the list of the pros. Indeed to make a portion of code testable we are often forced to restructure our code, but we can harm doing that.
```

Comment:
In fact, in order to make a part of the code testable we are often forced to restructure our code but this can be harmful.